src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -32,36 +28,24 @@ public LineService(LineRepository lineRepository, StationService stationService)
     public LineResponse saveLine(LineRequest request) {
         Station upStation = stationService.findById(request.getUpStationId());
         Station downStation = stationService.findById(request.getDownStationId());
-        Line persistLine = lineRepository.save(new Line(request.getName(), request.getColor(), upStation, downStation, request.getDistance()));
-        List<StationResponse> stations = getStations(persistLine).stream()
-                .map(it -> StationResponse.of(it))
-                .collect(Collectors.toList());
-        return LineResponse.of(persistLine, stations);
+        Line persistLine = lineRepository.save(request.toLine(upStation, downStation));
+        return LineResponse.of(persistLine, toStationResponses(persistLine));
     }
 
     public List<LineResponse> findLines() {
         List<Line> persistLines = lineRepository.findAll();
         return persistLines.stream()
-                .map(line -> {
-                    List<StationResponse> stations = getStations(line).stream()
-                            .map(it -> StationResponse.of(it))
-                            .collect(Collectors.toList());
-                    return LineResponse.of(line, stations);
-                })
+                .map(line -> LineResponse.of(line, toStationResponses(line)))
                 .collect(Collectors.toList());
     }
 
     public Line findLineById(Long id) {
         return lineRepository.findById(id).orElseThrow(RuntimeException::new);
     }
 
-
     public LineResponse findLineResponseById(Long id) {
         Line persistLine = findLineById(id);
-        List<StationResponse> stations = getStations(persistLine).stream()
-                .map(it -> StationResponse.of(it))
-                .collect(Collectors.toList());
-        return LineResponse.of(persistLine, stations);
+        return LineResponse.of(persistLine, toStationResponses(persistLine));
```

Comment:
LineResponse에 Line 을 전달하면 안에서 StationResponses를 만드는 형태도 좋지 않을까요...?
```suggestion
        return LineResponse.from(persistLine);
```
이 부분은 결국 같은 파라미터인데 두번 전달할 필요가 없지 않을까 싶어 말씀드립니다!

Reply:
말씀하신대로 파라미터가 뭔가 이상하네요 😭
바로 수정하겠습니다!